src/main/java/uk/gov/hmcts/ccd/domain/service/aggregated/DefaultGetDraftViewOperation.java

```diff
@@ -0,0 +1,102 @@
+import uk.gov.hmcts.ccd.domain.service.common.UIDService;
+import uk.gov.hmcts.ccd.domain.service.getcase.CreatorGetCaseOperation;
+import uk.gov.hmcts.ccd.domain.service.getcase.GetCaseOperation;
+
+import static uk.gov.hmcts.ccd.domain.model.aggregated.CaseViewTriggerBuilder.aCaseViewTrigger;
+import static uk.gov.hmcts.ccd.domain.model.definition.CaseDetailsBuilder.aCaseDetails;
+
+@Service
+@Qualifier(DefaultGetDraftViewOperation.QUALIFIER)
+public class DefaultGetDraftViewOperation extends AbstractDefaultGetCaseViewOperation implements GetCaseViewOperation {
+
+    public static final String QUALIFIER = "defaultDraft";
+    static final String DRAFT_ID = "DRAFT%s";
+    private static final CaseViewTrigger DELETE_TRIGGER = aCaseViewTrigger()
+        .withName("Delete")
+        .withDescription("Delete draft")
+        .withOrder(2)
+        .build();
+    private static final String RESUME = "Resume";
+
+    private final DraftGateway draftGateway;
+
+    @Autowired
+    public DefaultGetDraftViewOperation(@Qualifier(CreatorGetCaseOperation.QUALIFIER) final GetCaseOperation getCaseOperation,
+                                        final UIDefinitionRepository uiDefinitionRepository,
+                                        final CaseTypeService caseTypeService,
+                                        final UIDService uidService,
+                                        @Qualifier(DefaultDraftGateway.QUALIFIER) final DraftGateway draftGateway) {
+        super(getCaseOperation, uiDefinitionRepository, caseTypeService, uidService);
+        this.draftGateway = draftGateway;
+    }
+
+    @Override
+    public CaseView execute(String jurisdictionId, String caseTypeId, String draftId) {
+
+        final CaseType caseType = getCaseType(jurisdictionId, caseTypeId);
+        final DraftResponse draftResponse = draftGateway.get(draftId);
+
+        final CaseDetails caseDetails = buildCaseDetailsFromDraft(draftResponse);
+        final CaseViewTrigger resumeTrigger = buildResumeTriggerFromDraft(draftResponse);
+
+        final CaseTabCollection caseTabCollection = getCaseTabCollection(getCaseTypeIdFromDraft(draftResponse));
+
+        return merge(caseDetails, resumeTrigger, caseType, caseTabCollection);
+    }
+
+    private CaseViewTrigger buildResumeTriggerFromDraft(DraftResponse draftResponse) {
+        return aCaseViewTrigger()
+            .withId(draftResponse.getDocument().getEventTriggerId())
+            .withName(RESUME)
+            .withDescription(draftResponse.getDocument().getCaseDataContent().getEvent().getDescription())
+            .withOrder(1)
+            .build();
+    }
+
+    private String getCaseTypeIdFromDraft(DraftResponse draftResponse) {
+        return draftResponse.getDocument().getCaseTypeId();
+    }
+
+    private CaseView merge(CaseDetails caseDetails, CaseViewTrigger resumeTrigger, CaseType caseType, CaseTabCollection caseTabCollection) {
+        CaseView caseView = new CaseView();
```

Comment:
this logic it's similar to what we have in DefaultGetCaseViewOperation. 
It could be possible to introduce  a CaseViewBuilder and use that one in both DefaultGetCaseViewOperation and here?

Reply:
Let's keep with the no builders solution here:) We could add a tech debt story if needs be.